src/path-optimization/spline-gradient-based/cost.hh

```diff
@@ -40,41 +41,49 @@ namespace pathOptimization {
 /// The derivative of the cost is wrong when for freeflyer and planar
 /// joints. It lacks the derivative of the difference operator. The issue
 /// is that it is not a quadratic cost anymore.
-template <typename _Spline, int DerivativeOrder>
+template <typename _Spline>
 struct HPP_CORE_LOCAL SquaredLength  // : Cost
```

Comment:
Shouldn't we rename this class ?

Reply:
Doing so would break the API, but this could be included in the v5 release

Reply:
Any suggestion?

This won't break the api because it is internal to hpp-core so file.